src/kakarot/instructions/environmental_information.cairo

```diff
@@ -461,20 +462,27 @@ namespace EnvironmentalInformation {
         let (starknet_contract_address) = IRegistry.get_starknet_contract_address(
             contract_address=registry_address_, evm_contract_address=address_felt
         );
-        if (starknet_contract_address == 0) {
+        if (starknet_contract_address != 0) {
```

Comment:
I'm surprised that this works. It's surprising to me that the `ctx` variable assignment inside the scope of an if{} is persisted outside of it.

I would advise against this way of writing out the two possibilities for extcodesize, as it would not work in most languages

Reply:
I can change it back. The rationale was it allows gas to be updated in once place, instead of twice. Thought this would help avoid bugs.